File: frontend/src/api/stats.ts
import axiosInstance from './client';

export interface KPIStats {
  total_players: number;
  total_vips: number;
  total_mods: number;
  total_owners: number;
  total_banned: number;
  total_active: number;
  total_users: number;
  today_logins: number;
  bans_this_week: number;
  unbans_this_week: number;
}

export interface ActivityLog {
  id: number;
  user: string;
  action: string;
  resource: string;
  timestamp: string;
  details: string;
}

export interface TrendData {
  date: string;
  value: number;
}

export interface RoleDistribution {
  role: string;
  count: number;
}

export interface BanStats {
  total_banned: number;
  permanent_bans: number;
  temporary_bans: number;
  expiring_today: number;
  expiring_week: number;
  most_common_reason: string;
}

export interface DashboardOverview {
  kpi: KPIStats;
  recent_activities: ActivityLog[];
  role_distribution: RoleDistribution[];
  ban_stats: BanStats;
  player_trends: TrendData[];
  timestamp: string;
}

export const statsApi = {
  getKPIStats: async (): Promise<KPIStats> => {
    const response = await axiosInstance.get('/web/stats/kpi');
    return response.data;
  },

  getRecentActivity: async (limit = 20): Promise<ActivityLog[]> => {
    const response = await axiosInstance.get('/web/stats/activity', {
      params: { limit },
    });
    return response.data;
  },

  getPlayerTrends: async (days = 30): Promise<TrendData[]> => {
    const response = await axiosInstance.get('/web/stats/trends', {
      params: { days },
    });
    return response.data;
  },

  getRoleDistribution: async (): Promise<RoleDistribution[]> => {
    const response = await axiosInstance.get('/web/stats/roles');
    return response.data;
  },

  getBanStats: async (): Promise<BanStats> => {
    const response = await axiosInstance.get('/web/stats/bans');
    return response.data;
  },

  getDashboardOverview: async (): Promise<DashboardOverview> => {
    const response = await axiosInstance.get('/web/stats/overview');
    return response.data;
  },
};

File: internal/handler/stats.go
package handler

import (
	"context"
	"net/http"
	"strconv"
	"time"

	"github.com/angyxys/kexel/internal/service"
	"github.com/gin-gonic/gin"
)

type StatsHandler struct {
	serv *service.StatsService
}

func NewStatsHandler(serv *service.StatsService) *StatsHandler {
	return &StatsHandler{
		serv: serv,
	}
}

// GetKPIStats returns key performance indicators
func (h *StatsHandler) GetKPIStats(c *gin.Context) {
	ctx, cancel := context.WithTimeout(c, 10*time.Second)
	defer cancel()

	stats, err := h.serv.GetKPIStats(ctx)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{
			"message": "Failed to retrieve KPI stats",
			"status":  http.StatusInternalServerError,
		})
		return
	}

	c.JSON(http.StatusOK, stats)
}

// GetRecentActivity returns recent activities
func (h *StatsHandler) GetRecentActivity(c *gin.Context) {
	ctx, cancel := context.WithTimeout(c, 10*time.Second)
	defer cancel()

	limit, _ := strconv.Atoi(c.DefaultQuery("limit", "20"))
	if limit > 50 {
		limit = 50
	}

	activities, err := h.serv.GetRecentActivity(ctx, limit)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{
			"message": "Failed to retrieve recent activity",
			"status":  http.StatusInternalServerError,
		})
		return
	}

	c.JSON(http.StatusOK, activities)
}

// GetPlayerTrends returns player count trends
func (h *StatsHandler) GetPlayerTrends(c *gin.Context) {
	ctx, cancel := context.WithTimeout(c, 10*time.Second)
	defer cancel()

	days, _ := strconv.Atoi(c.DefaultQuery("days", "30"))
	if days > 365 {
		days = 365
	}
	if days < 7 {
		days = 7
	}

	trends, err := h.serv.GetPlayerTrends(ctx, days)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{
			"message": "Failed to retrieve trends",
			"status":  http.StatusInternalServerError,
		})
		return
	}

	c.JSON(http.StatusOK, trends)
}

// GetRoleDistribution returns role distribution
func (h *StatsHandler) GetRoleDistribution(c *gin.Context) {
	ctx, cancel := context.WithTimeout(c, 10*time.Second)
	defer cancel()

	distribution, err := h.serv.GetRoleDistribution(ctx)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{
			"message": "Failed to retrieve role distribution",
			"status":  http.StatusInternalServerError,
		})
		return
	}

	c.JSON(http.StatusOK, distribution)
}

// GetBanStats returns ban statistics
func (h *StatsHandler) GetBanStats(c *gin.Context) {
	ctx, cancel := context.WithTimeout(c, 10*time.Second)
	defer cancel()

	stats, err := h.serv.GetBanStats(ctx)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{
			"message": "Failed to retrieve ban stats",
			"status":  http.StatusInternalServerError,
		})
		return
	}

	c.JSON(http.StatusOK, stats)
}

// GetDashboardOverview returns comprehensive dashboard data
func (h *StatsHandler) GetDashboardOverview(c *gin.Context) {
	ctx, cancel := context.WithTimeout(c, 15*time.Second)
	defer cancel()

	kpiStats, _ := h.serv.GetKPIStats(ctx)
	activities, _ := h.serv.GetRecentActivity(ctx, 10)
	roleDistribution, _ := h.serv.GetRoleDistribution(ctx)
	banStats, _ := h.serv.GetBanStats(ctx)
	trends, _ := h.serv.GetPlayerTrends(ctx, 30)

	overview := gin.H{
		"kpi":                kpiStats,
		"recent_activities":  activities,
		"role_distribution":  roleDistribution,
		"ban_stats":          banStats,
		"player_trends":      trends,
		"timestamp":          time.Now(),
	}

	c.JSON(http.StatusOK, overview)
}
